Add outputs for the actual `google_sql_database_instance` resources

andreykaipov opened the pull request:
Addresses part of https://github.com/terraform-google-modules/terraform-google-sql-db/issues/192

andreykaipov commented:
Strange not sure what failed. I tested the mysql-ha example locally and it was ok. Will try to look at it next week

andreykaipov commented:
Yeah don't think it's anything related to this PR since my other PR #195 to fix the linting job also failed the integration tests.

morgante commented:
The integration test seems to be a timeout, we'll try re-running.

morgante commented:
/gcbrun

morgante commented:
We might need to increase the timeout for creating the instance:
```
$$$$$$ Running command `terraform apply -lock=true -lock-timeout=0s -input=false -auto-approve=true -no-color -parallelism=10 -refresh=true  ` in directory /workspace/test/fixtures/postgresql-public
       module.example.module.postgresql-db.random_id.suffix[0]: Creating...
       module.example.module.postgresql-db.random_id.suffix[0]: Creation complete after 0s [id=Htj0BQ]
       module.example.module.postgresql-db.null_resource.module_depends_on: Creating...
       module.example.module.postgresql-db.null_resource.module_depends_on: Creation complete after 0s [id=1790984371457975319]
       module.example.module.postgresql-db.google_sql_database_instance.default: Creating...
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [10s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [20s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [30s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [40s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [50s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [1m0s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [1m10s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [1m20s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [1m30s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [1m40s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [1m50s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [2m0s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [2m10s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [2m20s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [2m30s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [2m40s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [2m50s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [3m0s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [3m10s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [3m20s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [3m30s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [3m40s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [3m50s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [4m0s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [4m10s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [4m20s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [4m30s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [4m40s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [4m50s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [5m0s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [5m10s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [5m20s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [5m30s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [5m40s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [5m50s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [6m0s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [6m10s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [6m20s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [6m30s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [6m40s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [6m50s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [7m0s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [7m10s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [7m20s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [7m30s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [7m40s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [7m50s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [8m0s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [8m10s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [8m20s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [8m30s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [8m40s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [8m50s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [9m0s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [9m10s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [9m20s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [9m30s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [9m40s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [9m50s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [10m0s elapsed]
       module.example.module.postgresql-db.google_sql_database_instance.default: Still creating... [10m10s elapsed]
       
       Error: Error waiting for Create Instance: timeout while waiting for state to become 'DONE' (last state: 'RUNNING', timeout: 10m0s)
       
       
```

comment-bot-dev commented:
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/examples/mysql-ha/README.md /tmp/tmp.BkiwB9n0gE/generate_docs/workspace/examples/mysql-ha/README.md
34a35
> | instances | n/a |
37c38
< | replicas |  |
---
> | replicas | n/a |
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/examples/postgresql-ha/README.md /tmp/tmp.BkiwB9n0gE/generate_docs/workspace/examples/postgresql-ha/README.md
34a35
> | instances | n/a |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ```

morgante commented on modules/mssql/outputs.tf:
```diff
@@ -66,3 +66,9 @@ output "root_password" {
 }
+
+// Resources
+output "master" {
```
We're trying to stop using `master` in the module.

```suggestion
output "primary" {
```

andreykaipov replied:
Sorry for the slow response!

I've updated all the references where I could without making the diff too crazy. For example in the `safer_mysql` module the output is now `primary` but it still refers to `module.safer_mysql.master`.

morgante reviewed (changes requested):
Can you use `primary` to refer to the main instance? We're trying to stop using master terminology.

andreykaipov commented:
Okay I fixed everything up. Lint is passing, but integration is failing. Probably another timeout since it failed only after 9 minutes. On my local it took 18 minutes for a converge of `mysql-ha-local`, but I couldn't run a successful verify this time:

```
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Failed to complete #verify action: [cannot load such file -- google/apis/dns_v2beta1] on mysql-ha-local
```

Tried upgrading the test's `inspec-gcp` to `v0.23.4` but same thing. Upgrading to `v1.0.0` lets me runs the test, but with some different errors:

```
×  undefined method `item' for #<#<Class:0x0000556a6b84b2e8>::GoogleInSpec::SQL::Property::DatabaseInstanceSettings:0x0000556a6b2c49c8>
...
Test Summary: 26 successful, 25 failures, 0 skipped
```

Fortunately the new tests I added to the fixture do pass! Would you feel comfortable merging this in even though the integrations aren't passing? @morgante

morgante commented:
/gcbrun

andreykaipov commented:
Just saw that you merged this in! And even released it! Thanks so much! 😄 @morgante
